WordPress/Classes/ViewRelated/Plugins/PluginDirectoryViewModel.swift

```diff
@@ -34,73 +38,234 @@ class PluginDirectoryViewModel: Observable {
         }
     }
 
-    func tableViewModel(presenter: PluginPresenter & PluginListPresenter) -> ImmuTable {
+    func reloadFailed() {
+        allQueries
+            .filter { !isFetching(for: $0) }
+            .filter { !hasResults(for: $0) }
+            .flatMap { refreshAction(for: $0) }
+            .forEach { ActionDispatcher.dispatch($0) }
```

Comment:
I think the `isFetching` logic belongs in the Store, and IIRC it was already checking for that when processing refresh actions.